### PR TITLE
Add missing codecs for thrift generation

### PR DESCRIPTION
**Pull request by pkohan**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -2,13 +2,13 @@ name := "persist-json"
...
 
-version := "1.1.3"
+version := "1.2.0-WP"
 
 scalaVersion := "2.11.7"
 
 scalacOptions ++= Seq("-deprecation", "-unchecked")
 
-viewSettings
+//viewSettings
```
Comment (nvaidyanathan) [inline build.sbt:11]: Should we just remove this? Commented out code is dregs. We can always revert/add back if necessary.

File: src/main/scala/com/persist/JsonFormat.scala [hunk 2]
```diff
@@ -203,6 +203,10 @@ package object json {
       def read(json: Json): Set[T] = extractSeq[T](json).toSet
     }
 
+    implicit def setCollection[T: ReadCodec] = new ReadCodec[scala.collection.Set[T]] {
```
Comment (nvaidyanathan) [inline src/main/scala/com/persist/JsonFormat.scala:206]: This is the new feature we're adding? Do we have tests showing how this is used? What happens if I pass null? None?

File: src/main/scala/com/persist/JsonFormat.scala [hunk 3]
```diff
@@ -221,12 +225,22 @@ package object json {
...
+    }
+
     implicit def complexMap[K: ReadCodec, V: ReadCodec] = new ReadCodec[Map[K, V]] {
       def read(json: Json): Map[K, V] = castOrThrow(json).map { case (key, value) =>
         (implicitly[ReadCodec[K]].read(Json(key)), implicitly[ReadCodec[V]].read(value))
       }
     }
 
+    implicit def complexCollectionMap[K: ReadCodec, V: ReadCodec] = new ReadCodec[scala.collection.Map[K, V]] {
+      def read(json: Json): scala.collection.Map[K, V] = castOrThrow(json).map { case (key, value) =>
```
Comment (nvaidyanathan) [inline src/main/scala/com/persist/JsonFormat.scala:239]: What's thrown?

File: src/main/scala/com/persist/JsonFormat.scala [hunk 4]
```diff
@@ -239,7 +253,7 @@ package object json {
       }
     }
 
-    implicit def deriveHNil: ReadCodec[HNil] =
+    /*implicit def deriveHNil: ReadCodec[HNil] =
```
Comment (nvaidyanathan) [inline src/main/scala/com/persist/JsonFormat.scala:256]: Should we just remove this entirely?
